### PR TITLE
LVPN-10034: Cover ECH Remote Config option with tests

### PR DESCRIPTION
**Pull request by bbrutall**
- Implemented set of autotests to cover ECH feature

### CODE REVIEW

File: test/qa/test_remote_config.py [hunk 1]
```diff
@@ -832,3 +863,212 @@ def printdir():
...
+    assert not daemon_log_reader.wait_for_messages(
+        messages=RC_USE_LOCAL_CONFIG_MESSAGE,
+        cursor=cursor,
+    )
+
+    sh.nordvpn.connect()
+
+    assert daemon_log_reader.wait_for_messages(
+        messages=RC_USE_ECH_MESSAGE, cursor=cursor
+    ), f"Expected {RC_USE_ECH_MESSAGE} to be absent in the daemon log after reverting."
```
Comment (thegravi) [inline test/qa/test_remote_config.py]:
```suggestion
    ), f"Expected {RC_USE_ECH_MESSAGE} to be present in the daemon log after reverting."
```

File: test/qa/test_remote_config.py [hunk 2]
```diff
@@ -33,15 +33,46 @@
...
+    hash_path = f"{LOCAL_CACHE_DIR}/{RC_NORDWHISPER_HASH_FILE}"
+
+    with open(config_path, encoding="utf-8") as config_file:
+        config_data = json.load(config_file)
+
+    for config_entry in config_data.get("configs", []):
+        if config_entry.get("name") == "enable_ech":
+            for setting in config_entry.get("settings", []):
+                setting["value"] = enabled
+                setting["app_version"] = ">=4.4.0"
```
Comment (thegravi) [inline test/qa/test_remote_config.py:75]: Shall we guard for possible edge cases?
```suggestion
    modified = False
    for config_entry in config_data.get("configs", []):
        if config_entry.get("name") == "enable_ech":
            for setting in config_entry.get("settings", []):
                setting["value"] = enabled
                setting["app_version"] = ">=4.4.0"
                modified = True

    if not modified:
        raise ValueError(
            f"enable_ech config entry not found in {config_path}. "
            "Check if the remote config format has changed."
        )

```

File: test/qa/test_remote_config.py [hunk 3]
```diff
@@ -33,15 +33,46 @@
     f"[Info] feature [libtelio] config loaded from: {LOCAL_CACHE_DIR}",
     f"[Info] feature [nordvpn] config loaded from: {LOCAL_CACHE_DIR}",
 ]
```
Comment (thegravi) [inline test/qa/test_remote_config.py:38]: Missing `nordwhisper` here?

File: test/qa/test_remote_config.py (hunk 1 above)
Comment (bbrutall) [reply test/qa/test_remote_config.py]: fixed

File: test/qa/test_remote_config.py (hunk 2 above)
Comment (bbrutall) [reply test/qa/test_remote_config.py:75]: We have test that modifies local config files and verifies for mismatches. Also, added new tests `test_local_config_usage_missing_enable_ech_field`

File: test/qa/test_remote_config.py (hunk 3 above)
Comment (bbrutall) [reply test/qa/test_remote_config.py:38]: fixed

File: test/qa/test_remote_config.py [hunk 4]
```diff
@@ -832,3 +877,290 @@ def printdir():
...
+      9. Disconnect from VPN.
+      10. Disable 'use local RC config' mode.
+
+    :raises AssertionError: If the expected log message 'RC_USE_LOCAL_CONFIG_MESSAGE' is not found in the daemon log.
+    :raises AssertionError: If the expected missing-parameter warning message is not found in the daemon log.
+    :raises AssertionError: If the expected log message 'RC_USE_ECH_MESSAGE' is not found in the daemon log
+    """
+    rc_config_manager.set_permissions_cache_dir()
+
+    set_ech_for_local_config(enable_ech, entries_to_remove=[("name", "enable_ech")])
```
Comment (devzbysiu) [inline test/qa/test_remote_config.py:1133]: Does it make sense to pass `enable_ech` when this field config will be removed inside that `set_ech_for_local_config` function?

File: test/qa/test_remote_config.py [hunk 5]
```diff
@@ -832,3 +877,290 @@ def printdir():
...
+):
+    """
+    Test to verify that ECH is disabled via local RC usage and restored after disabling local RC usage.
+
+    Verify that disabling ECH via local RC config takes effect when local config usage is
+    enabled, and that it reverts to normal RC config behavior once local config usage
+    is disabled.
+
+    Test steps:
+      1. Set the local cached RC config to set 'enable_ech' to the parametrized value
```
Comment (devzbysiu) [inline test/qa/test_remote_config.py]: Copy-paste? There is no `enable_ech` parameter 😄

File: test/qa/test_remote_config.py [hunk 6]
```diff
@@ -832,3 +877,290 @@ def printdir():
...
+   """
+    set_technology_and_protocol(tech, proto, obfuscated)
+
+    sh.nordvpn.connect()
+
+    assert not daemon_log_reader.wait_for_messages(
+        messages=RC_USE_ECH_MESSAGE, cursor=daemon_log_cursor
+    ), f"Expected {RC_USE_ECH_MESSAGE} to be absent in the daemon log."
+
+    sh.nordvpn.disconnect()
```
Comment (devzbysiu) [inline test/qa/test_remote_config.py:914]: If the assertion above fails, we'll never call disconnect, right? Or is there some cleanup defined somewhere else? 🤔 
Same in other tests here

File: test/qa/test_remote_config.py [hunk 7]
```diff
@@ -26,22 +27,66 @@
...
+    for config_entry in config_data.get("configs", []):
+        if config_entry.get("name") == "enable_ech":
+            for setting in config_entry.get("settings", []):
+                setting["value"] = enabled
+                setting["app_version"] = ">=4.4.0"
+
+    for config_entry in config_data.get("configs", []):
+        for key, expected_value in entries_to_remove:
+            if config_entry.get(key) == expected_value:
+                config_entry.pop(key, None)
```
Comment (DoYouEvenCpp) [inline test/qa/test_remote_config.py:80]: [nit] wouldn't be more natural to keep those two for loops under the same context manager (` with open(config_path, encoding="utf-8") as config_file:`)?

File: test/qa/test_remote_config.py (hunk 5 above)
Comment (bbrutall) [reply test/qa/test_remote_config.py]: fixed

File: test/qa/test_remote_config.py (hunk 4 above)
Comment (bbrutall) [reply test/qa/test_remote_config.py:1133]: It doesn’t make much functional sense, but it’s still fine to pass it for testing consistency and to confirm the “missing field” behavior.

File: test/qa/test_remote_config.py (hunk 6 above)
Comment (bbrutall) [reply test/qa/test_remote_config.py:914]: There is a logout cleanup procedure in `nordvpnd_scope_function` fixture.

File: test/qa/test_remote_config.py (hunk 7 above)
Comment (bbrutall) [reply test/qa/test_remote_config.py:80]: Yes, it would be more natural to keep them together since both operations modify  the same config_data, but I separated them intentionally to keep the logic clearer (first apply the modifications, then perform optional deletions) and to avoid mixing two different behaviors in one loop.
